Modules/_csv.c

```diff
@@ -961,7 +987,8 @@ csv_reader(PyObject *module, PyObject *args, PyObject *keyword_args)
         Py_DECREF(self);
         return NULL;
     }
-    self->dialect = (DialectObj *)_call_dialect(dialect, keyword_args);
+    _csvstate *state = get_csv_state(module);
```

Comment:
state can be moved at the beginning of the function, to avoid calling get_csv_state() twice.